### ISSUE TITLE
Add documentation for combobox

### ISSUE THREAD

**Issue by Keboo** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#1177]
There are several additional features provided by the combo box that are not that easy to discover and should be documented.

- [ ] Add a page to the Wiki for ComboBox
- [ ] Add a link to it in the sidebar under Controls
- [ ] Document the two possible styles for ComboBox (`MaterialDesignComboBox` - this is the default, and `MaterialDesignFloatingHintComboBox`)
- [ ] Document that in addition to the default brushes on ComboBox there are a couple attached properties for changing colors. These are `TextFieldAssist.UnderlineBrush` for changing the color of the underline when the ComboBox is editable and the user is editing. `HintAssist.Foreground` for changing the color of the hint (when it is used).
- [ ] Document the `ComboBoxAssist.ClassicMode` attached property. When this is set to `true` it will cause the popup with the items to display strictly outside of the ComboBox rather than around it. This is more closely aligned with the default WPF ComboBox.

**Comment by Keboo** [comment]
Related #1171

**Comment by kevinmbuthi** [comment]
I agree

**Comment by pressacco** [comment]
Let me begin by saying that I am really impressed with this library. Thank you to all of those who contributed to the project!

---

It would be helpful if the documentation provided some guidance on how to implement the WPF default behavior for an editable `ComboBox`:

1. DO NOT display list of items when `ComboBox` is clicked.
   - The user can simply click the combo box and start typing without having the dropdown obstructing the UI.   
2. DO display list of items when `ComboBox` arrow is clicked.
   - When the user wants to see the list of options, they can explicitly request to see them by clicking the combo box arrow.

I am still relatively new to WPF, and the solution has escaped me.

### Additional Context

It appears that the *MaterialDesignInXaml* default behavior for a `IsEditable="True"` `ComboBox` is to:

- automatically display the list of dropdown items on `PreviewMouseDown` or `MouseLeftButtonDown`

**Comment by Keboo** [comment]
@Alex-451 yes, the wiki is just a special git repo within GitHub. I do keep an eye on the changes that people make to it and undo changes when needed, but most of the time when people make changes it is making things better, [so I have left it open to all edits](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/wiki/Contribution-Guidelines#wiki).
I have looked over your changes and they look great to me.

**Comment by Keboo** [comment]
If you are looking to get PR credit for hacktoberfest, feel free to submit a small PR to [this file](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/0584186b8aa85b0f167d5fe7511ee173be0bb314/MaterialDesign3.Demo.Wpf/Domain/MainWindowViewModel.cs) adding a wiki link to the new page. You can see an example of suck a link [here](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/dbe2a9686abef80a628270f21c2558c7117e6d45/MainDemo.Wpf/Domain/MainWindowViewModel.cs#L122)

**Comment by ArshadAriff** [comment]
hi @Keboo just wanted to know if i could work on this issue for HacktoberFest 2024!

**Comment by Keboo** [comment]
@ArshadAriff 
Yes absolutely!

**Comment by ArshadAriff** [comment]
@Keboo can you please eloborate on your comment
"If you are looking to get PR credit for hacktoberfest, feel free to submit a small PR to this file adding a wiki link to the new page. You can see an example of suck a link here"

😊

**Comment by ArshadAriff** [comment]
i should add a wiki to which page?

**Comment by Keboo** [comment]
@ArshadAriff Hacktoberfest only counts PRs to the main repository and not to the wiki. Create a new file inside of the [docs folder](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/tree/master/docs) and PR that. That way the PR will count.

**Comment by ArshadAriff** [comment]
Sir i have requested a PR please do merge sir!

**Comment by ArshadAriff** [comment]
@Keboo
